### ISSUE TITLE
Some libraries pkg-config files cannot link statically to libbrotlidec 

### ISSUE THREAD

**Issue by pypy702** [msys2/MINGW-packages#6769]
## Describe the bug
Libraries such as `libcurl` have a few dependencies in their `pkg-config` file. In this case, one of them is `-lbrotlidec`. This is fine when linking dynamically (`libbrotlidec.dll.a`), but when linking statically, you get errors like `ld: could not find -lbrotlidec`. This is because `libbrotlidec.a` doesn't exist. The real name is `libbrotlidec-static.a`

## Steps to Reproduce the Problem

1. In CMakeLists.txt write 

```
pkg_check_modules(LIBCURL REQUIRED libcurl)
target_include_directories(myPrj PRIVATE ${LIBCURL_INCLUDE_DIRS})
target_link_libraries(myPrj -static ${LIBCURL_STATIC_LIBRARIES}
```
2. Run `cmake .. -G 'MSYS Makefiles` from the build directory.

Expected behavior: The pkg-config file can locate `-lbrotlidec` statically.

## Additional Context: Operating System, Screenshots

- OS: [Windows 10 Pro version 2004 64bit]

**Comment by mati865** [comment]
I think it's duplicate of https://github.com/msys2/MINGW-packages/issues/3538 and should be reported upstream.

**Comment by pypy702** [comment]
Upstream meaning msys2/MSYS-packages?

**Comment by mati865** [comment]
Libcurl: https://github.com/curl/curl

**Comment by pypy702** [comment]
It's not a libcurl issue.
I created a file named `libcurl-static.pc` and linked to that.
```
#***************************************************************************
#                                  _   _ ____  _
#  Project                     ___| | | |  _ \| |
#                             / __| | | | |_) | |
#                            | (__| |_| |  _ <| |___
#                             \___|\___/|_| \_\_____|
#
# Copyright (C) 1998 - 2020, Daniel Stenberg, <daniel@haxx.se>, et al.
#
# This software is licensed as described in the file COPYING, which
# you should have received as part of this distribution. The terms
# are also available at https://curl.haxx.se/docs/copyright.html.
#
# You may opt to use, copy, modify, merge, publish, distribute and/or sell
# copies of the Software, and permit persons to whom the Software is
# furnished to do so, under the terms of the COPYING file.
#
# This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
# KIND, either express or implied.
#
###########################################################################

# This should most probably benefit from getting a "Requires:" field added
# dynamically by configure.
#
prefix=/mingw64
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include
supported_protocols="DICT FILE FTP FTPS GOPHER HTTP HTTPS IMAP IMAPS LDAP LDAPS POP3 POP3S RTSP SCP SFTP SMB SMBS SMTP SMTPS TELNET TFTP"
supported_features="SSL IPv6 libz brotli AsynchDNS IDN SSPI SPNEGO Kerberos NTLM TLS-SRP HTTP2 MultiSSL HTTPS-proxy"

Name: libcurl
URL: https://curl.haxx.se/
Description: Library to transfer files with ftp, http, etc.
Version: 7.71.0
Libs: -L${libdir} -lcurl 
Libs.private: -lnghttp2 -lidn2 -lssh2 -LD:/msys64/mingw64/lib -lssh2 -lpsl -lssl -lcrypto -lssl -lcrypto -lgdi32 -lcrypt32 -lwldap32 -lbrotlidec-static -lbrotlicommon-static -lz -lws2_32
Cflags: -I${includedir} 
```
This time, it could find and link to libbrotlidec fine. But then I got errors with undefined reference like this (`D:/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/msys64/mingw64/lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../lib\libcurl.a(libcurl_la-http2.o):(.text+0x31): undefined reference to `__imp_nghttp2_session_del'`). These are all `nghttp2` errors. So I extracted `libcurl.a` and `libcurl.dll.a` into `/mingw64/lib` from the [official cURL website](curl.haxx.se), and it worked.
So it's a `MINGW-packages` bug.

**Comment by mingwandroid** [comment]
You need to collect all the transitively used static libraries into each link confab line.

It is an upstream issue if the .pc file is either not created or, as in the case of your hand crafted one, incorrect. It is not the job of distributions to craft these build system metadata files, it is up to the upstream projects' to manage this and to make sure their build systems emit them correctly as appropriate, mostly *specifically because* those are the people who know the details about how their software needs to be linked.

This should be closed IMHO.

**Comment by pypy702** [comment]
Yep.
It's not built properly for static linking.
The pc file is only for dynamic, and "-DCURL_STATIC_LIB" is not in the CFLAGS.
And it is not linked properly.
Is it because it's built with "--enable-static" and "--enable-shared" at the same time?

**Comment by Tearnote** [comment]
I just ran into this issue compiling with freetype statically, which depends on libbrotli.

No need to use CMake to reproduce. Output of `freetype-config --static --libs`:
```-LC:/msys64/mingw64/lib -lfreetype -lbz2 -lpng16 -lz -lharfbuzz -lm -lusp10 -lgdi32 -lrpcrt4 -ldwrite -lgraphite2 -lglib-2.0 -lintl -lws2_32 -lole32 -lwinmm -lshlwapi -pthread -lm -lpcre -lbrotlidec -lbrotlicommon```

This does not work because there exists only `-lbrotlidec-static` and `-lbrotlicommon-static`. To me, it appears to be very much a packaging/configuration issue.

**Comment by mati865** [comment]
@Tearnote it's upstream issue: https://github.com/google/brotli/issues/795

**Comment by jay** [comment]
It looks like someone added a workaround for this in mingw-w64-brotli

https://github.com/msys2/MINGW-packages/blob/ee1f4bf7c82f34fe64a69f30cb5ce303a8685afd/mingw-w64-brotli/PKGBUILD#L67-L70

**Comment by lazka** [comment]
yeah, a basic brotlidec static example builds fine with pkgconf now. Feel free to open a new issue if there is anything missing still.
